content/docs/operating/security.md

```diff
@@ -177,6 +182,38 @@ and bandwidth.
 It is recommended to monitor all components for failure, and to have them
 automatically restart on failure.
 
+## Server-side TLS and authentication
+
+In the future, server-side TLS support will be rolled out to the different
+projects under the Prometheus organization. Those projects include Prometheus,
+Alertmanager, Pushgateway and the official exporters.
+
+The golang projects will share the same TLS library, which will be based on the
+Golang vanilla TLS library. We keep [Golang's default TLS
+parameters](https://golang.org/pkg/crypto/tls/#Config), with one exception: we
+support only TLSv1.2 and higher.
+
+Research will be made later on the non-golang exporters to see how to best
+address server-side TLS, but you can expect a policy close to this one.
+
+The TLS support on the projects should enable you to build a secure tunnel
+between the servers and your reverse proxies if you have more specialized TLS
+needs, like a cipher suite or older TLS version.
+
+Server-side Basic Authentication is also supported. In such a case, usernames
+and passwords are provided in the same configuration file as the TLS
+configuration. Basic Authentication can be used without TLS, but it will then
+expose usernames and passwords over the network.
+
+Passwords are hashed with the [bcrypt](https://en.wikipedia.org/wiki/Bcrypt)
+algorithm, and it is your responsibility to pick the number of rounds that
+matches your security standards. More rounds make brute-force more complicated
+at the cost of more CPU power and more time to authenticate the requests.
+
+The web security configuration file, with users and path to certificates, can be
+changed on the fly, with the exception of enabling/disabling TLS entirely. It is
+your responsibility to protect that file from unwanted reads/writes.
```

Comment:
Hmm, this final bit would be better covered up in the Secrets section I think.

Reply:
moved